ginga/misc/plugins/ThumbsBase.py

```diff
@@ -127,6 +129,12 @@ def add_image(self, viewer, chname, image):
             self.thumb_generator.set_image(image)
             imgwin = self.thumb_generator.get_image_as_widget()
 
+        if not self.settings.get('label_name', True):
+            thumbnamekey = self.settings.get('mouseover_name_key', 'NAME')
+            self.keywords.insert(0, thumbnamekey)
```

Comment:
I think `self.keywords` is supposed to be a static list.  If you keep inserting here at this point in the code, I think the list will keep growing.  Maybe do it once in `__init__`?